crates/node/src/command.rs

```
@@ -52,69 +52,138 @@ impl SubstrateCli for Cli {
+            for chain_spec in madara_configs.chain_specs {
+                configs::fetch_and_validate_file(
+                    madara_configs.remote_base_path.clone(),
+                    chain_spec,
+                    madara_path.clone() + "/configs/chain-specs/",
+                    cli.run.update_configs,
                 )?;
             }
+            cli.run.run_cmd.shared_params.chain = Some(madara_path + "/chain-specs/testnet-sharingan-raw.json");
         }
+        _ => {}
     }
 
-    // TODO confirm with the CI that we are fetching all and fetch dynamically
-    // Issue #982
-    for file in constants::CAIRO_CONTRACTS_FILES {
-        let src_path = utils::get_project_path();
-        if let Ok(src_path) = src_path {
-            let src_path = src_path + "/configs/cairo-contracts/" + file;
-            utils::copy_from_filesystem(src_path, madara_path.clone() + "/cairo-contracts")?;
-        } else {
-            utils::fetch_from_url(
-                constants::CAIRO_CONTRACTS_URL.to_string() + file,
-                madara_path.clone() + "/cairo-contracts",
+    if cli.run.run_cmd.shared_params.chain.is_some() {
+        cli.run.run_cmd.rpc_external = true;
+        cli.run.run_cmd.rpc_methods = RpcMethods::Unsafe;
+    }
+
+    Ok(())
+}
+
+fn fetch_madara_configs(cli: &mut Cli) -> Result<(), String> {
+    let madara_path = cli.run.madara_path.clone().expect("Failed retrieving madara_path").to_str().unwrap().to_string();
+    let local_path = utils::get_project_path();
+
+    if let Some(configs_url) = cli.run.configs_url.clone() {
+        utils::fetch_from_url(configs_url, madara_path.clone() + "/configs", cli.run.update_configs)?;
+    } else if let Ok(ref src_path) = local_path {
+        let src_path = src_path.clone() + "/configs/index.json";
+        utils::copy_from_filesystem(src_path, madara_path.clone() + "/configs", cli.run.update_configs)?;
+    } else if !cli.run.disable_url_fetch {
```

What happend at this point if `cli.run.disable_url_fetch` is true. Does madara still work ? If not we should return an error message enjoining the use to pass one of the 3 required arg

---

My understanding is that we will panic at l158 `.expect("Failed to load madara configs");`
I would rather return an error earlier if madara can't work with this combination of args

---

the configs can be manually added in the madara path, or they can be already there from a previous run